crates/router/src/db/merchant_key_store.rs

```diff
@@ -128,6 +135,32 @@ impl MerchantKeyStoreInterface for Store {
             .await
         }
     }
+
+    #[cfg(feature = "olap")]
+    async fn list_multiple_key_stores(
+        &self,
+        merchant_ids: Vec<String>,
+        key: &Secret<Vec<u8>>,
+    ) -> CustomResult<Vec<domain::MerchantKeyStore>, errors::StorageError> {
+        let fetch_func = || async {
+            let conn = connection::pg_connection_read(self).await?;
+
+            diesel_models::merchant_key_store::MerchantKeyStore::list_multiple_key_stores(
+                &conn,
+                merchant_ids,
+            )
+            .await
+            .map_err(Into::into)
+            .into_report()
+        };
+
+        futures::future::try_join_all(fetch_func().await?.into_iter().map(|x| async {
+            x.convert(key)
```

Comment:
Would have preferred a better variable name instead of `x`: `merchant_key_store` or `key_store` perhaps?